### PR TITLE
Rewrite EFA to jaspresults

### PR DESCRIPTION
**Pull request by vankesteren**
Also included several improvements, such as:
- Different factor extraction methods ([#335](https://github.com/jasp-stats/jasp-issues/issues/335), [#89](https://github.com/jasp-stats/jasp-issues/issues/89))
- Kaiser-Meyer-Olkin test and Bartlett's test of sphericity is included as an assumption check ([#79](https://github.com/jasp-stats/jasp-issues/issues/79))

### CODE REVIEW

File: JASP-Tests/R/tests/figs/ExploratoryFactorAnalysis/exploratoryfactoranalysis-path-diagram.svg [hunk 1]
```diff
@@ -1,42 +1 @@
...
-<text x='158.74' y='295.48' style='font-size: 21.76px; font-family: Liberation Sans;' textLength='25.38px' lengthAdjust='spacingAndGlyphs'>F1</text>
-<text x='460.59' y='488.86' style='font-size: 25.77px; font-family: Liberation Sans;' textLength='107.39px' lengthAdjust='spacingAndGlyphs'>contWide</text>
-<text x='464.87' y='360.86' style='font-size: 25.77px; font-family: Liberation Sans;' textLength='98.83px' lengthAdjust='spacingAndGlyphs'>contcor1</text>
-<text x='472.77' y='232.86' style='font-size: 25.77px; font-family: Liberation Sans;' textLength='83.03px' lengthAdjust='spacingAndGlyphs'>facFifty</text>
-<text x='453.40' y='104.86' style='font-size: 25.77px; font-family: Liberation Sans;' textLength='121.77px' lengthAdjust='spacingAndGlyphs'>contExpon</text>
 </svg>
```
Comment (TimKDJ) [inline JASP-Tests/R/tests/figs/ExploratoryFactorAnalysis/exploratoryfactoranalysis-path-diagram.svg:94]: This is not right.. where is the reference svg?

File: JASP-Engine/JASP/R/exploratoryfactoranalysis.R [hunk 2]
```diff
@@ -15,850 +15,572 @@
...
+}
+
+.efaBartlett <- function(modelContainer, dataset, options, ready) {
+  if (!options[["bartest"]] || !is.null(modelContainer[["bartab"]])) return()
+  
+  bartab <- createJaspTable("Bartlett's test")
+  bartab$dependOn("bartest")
+  bartab$addColumnInfo(name = "chisq", title = "\u03a7\u00b2", type = "number", format = "dp:3")
+  bartab$addColumnInfo(name = "df", title = "df", type = "number", format = "dp:3")
+  bartab$addColumnInfo(name = "pval", title = "p-value", type = "number", format = "dp:3;p:.001")
```
Comment (TimKDJ) [inline JASP-Engine/JASP/R/exploratoryfactoranalysis.R]:
```suggestion
  bartab$addColumnInfo(name = "pval", title = "p", type = "number", format = "dp:3;p:.001")
```
to be in-line with your other table (and most of jasp I think)

**Review by TimKDJ — CHANGES_REQUESTED** [review]
So the code looks very clean -- nice!
Two code comments and some issues in a .jasp file:

[problemsInEFA.jasp.zip](https://github.com/jasp-stats/jasp-desktop/files/3792948/problemsInEFA.jasp.zip)

**Comment by vankesteren** [comment]
Alright, fixes done!

**Comment by TimKDJ** [comment]
Alright look good to me. The only thing is the failing scree plot. I saw you put a skip there for the PCA, so I'm guessing that's also required here.

**Comment by vankesteren** [comment]
You're right! For the scree plot data is generated. Fixed.

**Pull request merged**